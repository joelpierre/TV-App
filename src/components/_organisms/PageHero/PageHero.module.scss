.PageHero {
  position: relative;
  min-height: 260px;
  padding: spacing(4) 0;

  &::after {
    position: absolute;
    content: '';
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: -100;
    background-color: $color-brand;
  }

  @include breakpoint('sm') {
    height: 430px;
    padding: spacing(12) 0;

    &::after {
      height: 580px;
    }
  }

  &__container {
    @include max-width-container;
  }

  &__content {
    display: flex;
    flex-direction: column;
  }

  &__heading {
    @include type-heading-1;
    margin-bottom: spacing(4);
    color: $color-tint-alpha;
    font-size: 30px;
  }

  &__copy {
    font-size: 30px;
    color: $color-tint-beta;
  }

  &__heading,
  &__copy {
    @include breakpoint('sm') {
      max-width: 80%;
    }
  }
}
