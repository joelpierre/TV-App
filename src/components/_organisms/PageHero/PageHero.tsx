import classNames from 'classnames';
import React from 'react';

import styles from './PageHero.module.scss';

interface IPageHeroProps {
  className?: string;
}

const PageHero: React.FunctionComponent<IPageHeroProps> = ({
  className
}) => {
  return (
    <article className={classNames(styles.PageHero, className)}>
      <div className={styles.PageHero__container}>
        <div className={styles.PageHero__content}>
          <h1 className={styles.PageHero__heading}>
            TV Bland
          </h1>
          <p className={styles.PageHero__copy}>
            TV Show and web series database.<br />Create personalised schedules.
            Episode guide, cast, crew and character information.
          </p>
        </div>
      </div>
    </article>
  );
};

export default PageHero;
