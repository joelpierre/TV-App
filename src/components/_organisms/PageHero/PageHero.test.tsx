import React from 'react';
import { render, RenderResult } from 'test-utils';

import PageHero from './PageHero';

const setup = (props = {}): RenderResult => {
  const setupProps = { ...props };
  return render(<PageHero {...setupProps} />);
};

describe('<PageHero/>', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Should render without crashing', () => {
    expect(wrapper.container).not.toBeEmptyDOMElement();
  });

  it('should render a H1 tag', () => {
    expect(wrapper.container.getElementsByTagName('h1')).toHaveLength(1);
  });

  it('should match snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
