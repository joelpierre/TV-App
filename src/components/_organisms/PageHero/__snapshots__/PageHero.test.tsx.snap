// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHero/> should match snapshot 1`] = `
<DocumentFragment>
  <article
    class="PageHero"
  >
    <div
      class="PageHero__container"
    >
      <div
        class="PageHero__content"
      >
        <h1
          class="PageHero__heading"
        >
          TV Bland
        </h1>
        <p
          class="PageHero__copy"
        >
          TV Show and web series database.
          <br />
          Create personalised schedules. Episode guide, cast, crew and character information.
        </p>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
